ci/script/clone_all_rspec_repos

```diff
@@ -9,11 +9,8 @@ if is_mri; then
   clone_repo "rspec-core"
   clone_repo "rspec-expectations"
   clone_repo "rspec-mocks"
-  clone_repo "rspec-rails"
-
-  if rspec_support_compatible; then
-    clone_repo "rspec-support"
-  fi
+  clone_repo "rspec-rails" "4-0-dev"
```

Comment:
Maybe we should introduce a `maintenance-rspec-rails-branch` file?

Still, that `4-0-dev` branch has nothing to do with `rspec-rails` 4.0, but rather it's there to adjust `rspec-rails` dependencies so they point back to the right version of the current repository where it's tested in a sub-build.

Reply:
The `rspec-rails` branch you're looking for is `main` for `4-0-dev`, if you'd like to build out the file based approach and use it for both branches I'd be happy with that.

Reply:
`main` doesn't work, because it [has `main`](https://github.com/rspec/rspec-rails/blob/9217638aff96df56753eeafe60e16bc4c35b8566/maintenance-branch#L1) in it, this means that other `rspec-*` projects will be used for `rspec-rails` sub build. `4-0-dev` [points to](https://github.com/rspec/rspec-rails/blob/69e23e20166f704a52e30d9b1a3a33d608def353/maintenance-branch#L1) `4-0-dev` branch, and `rspec-*` will be used from it.

I think we can live with temporarily having a `4-0-dev` branch in `rspec-rails` and reverting this line here when we do a release and `4-0-dev` is merged to `main`.

Reply:
> main doesn't work, because it has main in it, this means that other rspec-* projects will be used for rspec-rails sub build. 4-0-dev points to 4-0-dev branch, and rspec-* will be used from it.

No each project has `4-0-dev` in and points to `rspec-rails` main, thats what this function does, it overrides the `4-0-dev` in the local file.

Reply:
Oh, I see! It doesn't really matter what is in `maintenance-branch` of the cloned `rspec-rails`, it's unused in sub-builds.
We check [if there are necessary repos in sibling directories](
https://github.com/rspec/rspec-rails/blob/9217638aff96df56753eeafe60e16bc4c35b8566/Gemfile-rspec-dependencies#L4) and use them when bundling.
This slipped my mind completely.

There was a strange build failure that `bundler` cannot find `rspec-rails`'s `4.0.0.pre` `rspec-support` dependency in any of the sources. Might have been temporary.

I intend to:
 - delete `4-0-dev` `rspec-rails` branch
 - change to `clone_repo "rspec-rails" "main"` here and in generated PRs

Thanks for the heads-up 👍

Reply:
Found it. The failure was because `RSPEC_CI: true` was not set.